### PR TITLE
CDAP-1628 add cdap version as a property to all hive tables in

### PR DESCRIPTION
**Pull request by albertshau**
preparation for upgrade work. Also did some refactoring to move
logic around building the create statement for hive tables into
one class.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT1021

File: cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java [hunk 1]
```diff
@@ -105,26 +101,25 @@ public void enableStream(HttpRequest request, HttpResponder responder,
...
         }
-        streamLocationURI = streamLocation.toURI().toString();
       } catch (IOException e) {
         LOG.info("Could not find stream {} to enable explore on.", streamName, e);
         responder.sendString(HttpResponseStatus.NOT_FOUND, "Could not find stream " + streamName);
         return;
       }
 
-      LOG.debug("Enabling explore for stream {} at location {}", streamName, streamLocationURI);
+      LOG.debug("Enabling explore for stream {} at location {}", streamName, streamLocation);
```
Comment (bdmogal) [inline cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java]: I remember seeing ``Location.toString()`` returning the object reference as output. Can you please check?

**Comment by bdmogal** [comment]
Should we also change the table names for streams and datasets or is that in a different PR?

**Comment by bdmogal** [comment]
Just 1 minor comment and one question. LGTM otherwise.

File: cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java (hunk 1 above)
Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java]: o yes changed the type without changing this

**Comment by albertshau** [comment]
names will be done in a separate PR

**Pull request merged**